applications/nrf_desktop/src/hw_interface/passkey_buttons.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#define MODULE passkey
+#include "module_state_event.h"
+
+#include "button_event.h"
+#include "passkey_event.h"
+
+#include "passkey_buttons_def.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_PASSKEY_LOG_LEVEL);
+
+#define CLEAR_INPUT_ON_HOLD_TIMEOUT K_SECONDS(2)
+
+enum state {
+	STATE_DISABLED,
+	STATE_IDLE,
+	STATE_ACTIVE,
+};
+
+struct input_buf {
+	u8_t buf[CONFIG_DESKTOP_PASSKEY_MAX_LEN];
+	u8_t cnt;
+};
+
+static enum state state;
+static struct input_buf digits;
+
+
+static u32_t encode_input(struct input_buf *buffer)
+{
+	u32_t res = 0;
+	u32_t mult = 1;
+
+	for (size_t i = buffer->cnt; i > 0; i--) {
+		res += buffer->buf[i - 1] * mult;
+		mult *= DIGIT_CNT;
+	}
```

Comment:
Please add a comment explaining the algorithm. e.g.

> translate the buffer input "00123" into the u32_t value 123.

Reply:
ok